roxie/roxiemem/roxiemem.cpp

```diff
@@ -2911,13 +2916,35 @@ class CRoxieDirectFixedBlockedRowHeap : public CRoxieDirectFixedRowHeap<T>
     {
     }
 
+    ~CRoxieDirectFixedBlockedRowHeap()
+    {
+#if defined(_DEBUG)
+        if (curRow != numRows)
+            DBGLOG("Blocked Row Heap still contains %u rows", numRows - curRow);
+#endif
+    }
+
     virtual void *allocate()
     {
+#if defined(CHECK_FOR_BLOCKED_MULTI_THREAD)
+        if (active++ != 0)
+        {
+            active--;
+            throw makeStringException(ROXIEMM_BLOCKED_MULTI_THREAD, "Multiple concurrent calls to CRoxieDirectFixedBlockedRowHeap::allocate");
+        }
+        //Potentially yield to make it more likely that another thread will call allocate() before this thread exits
+        MilliSleep(0);
```

Comment:
Do you want this before the throw ?

Reply:
It doesn't need to be - if the throw is happening then the problem will be caught.